plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -108,6 +114,47 @@ public ElasticsearchMetadata(TypeManager typeManager, ElasticsearchClient client
         this.schemaName = config.getDefaultSchema();
     }
 
+    private static boolean supportsPredicates(IndexMetadata.Type type)
+    {
+        if (type instanceof DateTimeType) {
+            return true;
+        }
+
+        if (type instanceof PrimitiveType) {
+            switch (((PrimitiveType) type).getName().toLowerCase(ENGLISH)) {
+                case "boolean":
+                case "byte":
+                case "short":
+                case "integer":
+                case "long":
+                case "double":
+                case "float":
+                case "keyword":
+                    return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static boolean isPassthroughQuery(ElasticsearchTableHandle table)
+    {
+        return table.getType().equals(QUERY);
+    }
+
+    private static ElasticsearchColumnHandleProperty generateHandleProperties(IndexMetadata.Field field)
+    {
+        Map<String, ElasticsearchColumnHandleProperty> children = ImmutableMap.of();
+
+        if (field.getType() instanceof IndexMetadata.ObjectType) {
```

Comment:
@phd3 the `asRawJson` value for `col0.B` would be `true`.

```
ESProps(
    col0
    <A : ESProps(A, [], false)>, B : ESProps(B, [], true)>
    false);
```

I think I know what you're bringing up though. Let's look at a more complete example just to make sure we're in sync.

If we have the following elasticsearch mapping

```
{
  "_meta": {
    "presto": {
      "es_array_object": {
        "isArray": true,
        "array_of_string_arrays": {
          "asRawJson": true
        },
        "array_of_int_arrays": {
          "asRawJson": true
        }
      },
      "es_raw_object": {
        "asRawJson": true,
        "array_of_string_arrays": {
          "isArray": true
        },
        "array_of_int_arrays": {
          "isArray": true
        }
      }
    }
  },
  "properties": {
    "es_array_object": {
      "type": "object",
      "properties": {
        "array_of_string_arrays": {
          "type": "keyword"
        },
        "array_of_int_arrays": {
          "type": "integer"
        }
      }
    },
    "es_raw_object": {
      "type": "object",
      "properties": {
        "array_of_string_arrays": {
          "type": "keyword"
        },
        "array_of_int_arrays": {
          "type": "integer"
        }
      }
    }
  }
}
```

The two resulting ESProps will be:
```
ESProps(
    es_array_object,
    [<array_of_string_arrays : ESProps(array_of_string_arrays, [], true)>,
     <array_of_int_arrays : ESProps(array_of_int_arrays, [], true)>],
    false);
```

```
ESProps(
    es_raw_object,
    [<array_of_string_arrays : ESProps(array_of_string_arrays, [], false)>,
     array_of_int_arrays : ESProps(array_of_int_arrays, [], false)>],
    true);
```

What I think you're pointing out is that it's strange that despite the fact that we have `es_raw_object` that contains two children entries that won't be used since the parent is set to `asRawJson` whereas in the `es_array_object` case, we will use the children. Is that correct?